pkg/spoke/controllers/finalizercontroller/appliedmanifestwork_finalize_controller.go

```diff
@@ -48,6 +50,9 @@ func NewAppliedManifestWorkFinalizeController(
 	return factory.New().
 		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
 			accessor, _ := meta.Accessor(obj)
+			if !strings.HasPrefix(accessor.GetName(), hubHash) {
+				return ""
```

Comment:
we have a queueKeyFunc in helper that you can use. Also I think we should use WithFilteredInformersQueueKeyFunc

Reply:
> 
Is the queueKeyFunc in helper this func？
https://github.com/open-cluster-management-io/work/blob/8f206974ddc0bb239e014768dbae19d0f37cad74/pkg/helper/helpers.go#L322
This func will return a manifestwork key, but in this controller we need a appliedmanifestwork key. May I can provide another func in helper?

Reply:
> ok, that will be great. And maybe also a filter func in helper that can be reused by multiple controllers.

The changes have been committed